recipes/ginkgo/all/conanfile.py

```diff
@@ -101,8 +101,51 @@ def build_requirements(self):
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def _get_visual_generator(self):
+        compiler_version = str(self.settings.compiler.version)
+        if self.settings.compiler == "msvc":
+            compiler_version = {
+                "170": "11",
+                "180": "12",
+                "190": "14",
+                "191": "15",
+                "192": "16",
+                "193": "17",
+            }[compiler_version]
+
+        visual_gen_suffix = {
+            "8": "8 2005",
+            "9": "9 2008",
+            "10": "10 2010",
+            "11": "11 2012",
+            "12": "12 2013",
+            "14": "14 2015",
```

Comment:
Note that we've never tested, and don't really support those compilers.

Reply:
Yes I know, it's just generic code. We raise if Visual Studio < 2019 anyway.

Reply:
Thanks for double-checking @upsj :)